app/Http/Controllers/InternetController.php

```diff
@@ -125,7 +118,7 @@ public function addMacAddress(Request $request)
 
         $macAddress = new MacAddress();
         $macAddress->user_id = Auth::user()->id;
-        if (Auth::user()->isAdmin() && $request->has('user_id')) { // TODO: use gate
+        if (Auth::user()->can('accept', $macAddress) && $request->has('user_id')) {
```

Comment:
Should not this be a creation instead of an accept permission?

Reply:
I thought of this first, but according to the logic here, anyone can create a mac address (with access to the internet module), but only admins can accept the opened request.
So what I just changed here is to separate the mac address accepting permission from the global admin role and created an 'accept' gate.